website/addons/base/generic_views.py

```diff
@@ -0,0 +1,138 @@
+        return serializer.serialized_user_settings
+    _account_list.__name__ = '{0}_account_list'.format(addon_short_name)
+    return _account_list
+
+def folder_list(addon_short_name, addon_full_name, get_folders):
+    @must_have_addon(addon_short_name, 'node')
+    @must_be_addon_authorizer(addon_short_name)
+    def _folder_list(node_addon, **kwargs):
+        """Returns a list of folders"""
+        if not node_addon.has_auth:
+            raise HTTPError(http.FORBIDDEN)
+
+        folder_id = request.args.get('folderId')
+        return get_folders(node_addon, folder_id)
+    _folder_list.__name__ = '{0}_folder_list'.format(addon_short_name)
+    return _folder_list
+
+def root_folder(addon_short_name):
+    def _root_folder(node_settings, auth, **kwargs):
+        """Return the Rubeus/HGrid-formatted response for the root folder only."""
+        # Quit if node settings does not have authentication
+        if not node_settings.has_auth or not node_settings.folder_id:
+            return None
+        node = node_settings.owner
+        root = rubeus.build_addon_root(
+            node_settings=node_settings,
+            name=node_settings.fetch_folder_name(),
+            permissions=auth,
+            nodeUrl=node.url,
+            nodeApiUrl=node.api_url,
+        )
+        return [root]
+    _root_folder.__name__ = '{0}_root_folder'.format(addon_short_name)
+    return _root_folder
+
+
+def get_config(addon_short_name, Serializer):
+    @must_be_logged_in
+    @must_have_addon(addon_short_name, 'node')
+    @must_be_valid_project
+    @must_have_permission(permissions.WRITE)
+    def _get_config(node_addon, auth, **kwargs):
+        """API that returns the serialized node settings."""
+        return {
+            'result': Serializer().serialize_settings(
+                node_addon,
+                auth.user
+            )
+        }
+    _get_config.__name__ = '{0}_get_config'.format(addon_short_name)
+    return _get_config
+
+def set_config(addon_short_name, addon_full_name, set_folder):
+    @must_not_be_registration
+    @must_have_addon(addon_short_name, 'user')
+    @must_have_addon(addon_short_name, 'node')
+    @must_be_addon_authorizer(addon_short_name)
+    @must_have_permission(permissions.WRITE)
+    def _set_config(node_addon, user_addon, auth, **kwargs):
+        """View for changing a node's linked folder."""
+        folder = request.json.get('selected')
+        set_folder(node_addon, folder, auth)
+
+        path = folder['path']
+        return {
+            'result': {
+                'folder': {
+                    'name': path.replace('All Files', '') if path != 'All Files' else '/ (Full {0})'.format(
+                        addon_full_name
+                    ),
+                    'path': path,
+                }
+            },
+            'message': 'Successfully updated settings.',
+        }
+    _set_config.__name__ = '{0}_set_config'.format(addon_short_name)
+    return _set_config
```

Comment:
Should this function still return `urls` in addition to the `folder`? It looks like `dropbox_config_put` used to do that, and `submitSettings` in `folderPickerNode.js` is still expecting `urls` This is probably related to not being able to successfully set/change the dropbox and box folders.